doc/api/n-api.md

```diff
@@ -118,6 +118,89 @@ must make use exclusively of N-API by restricting itself to using
 and by checking, for all external libraries that it uses, that the external
 library makes ABI stability guarantees similar to N-API.
 
+## Building
+
+Unlike addons coded in JavaScript, developing and deploying Node native
+addons using N-API requires an additional set of tools. Besides the basic
+tools required to develop for Node, the native addon developer requires a
+toolchain that can compile C and C++ code into a binary. In addition,
+depending upon how the native addon is deployed, the *user* of the native
+addon will also need to have a C/C++ toolchain installed.
+
+For Linux developers, the necessary C/C++ toolchain is typically already
+installed. For systems lacking the necessary toolchain, the
+[LLVM](https://llvm.org) compiler infrastructure is a good choice.
+
+For Mac developers, [Xcode](https://developer.apple.com/xcode/) offers all
+the required compiler tools. However, it is not necessary to install the
+entire Xcode IDE. The following command installs the necessary toolchain:
+
+```bash
+xcode-select --install
+```
+
+For Windows developers, [Visual Studio](https://visualstudio.microsoft.com)
+offers all the required compiler tools. However, it is not necessary to
+install the entire Visual Studio IDE. The following command installs the
+necessary toolchain:
+
+```bash
+npm install --global --production windows-build-tools
+```
+
+The sections below describe the additional tools available for developing
+and deploying Node native addons.
+
+### Build tools
+
+> Both the tools listed here require that *users* of the native
+> addon have a C/C++ toolchain installed in order to successfully install
+> the native addon.
+
+#### node-gyp
+
+[node-gyp](https://github.com/nodejs/node-gyp) is a build system based on
+Google's [GYP](https://gyp.gsrc.io) tool and comes bundled with npm. GYP,
+and therefore node-gyp, requires that Python be installed.
+
+Historically, node-gyp has been the tool of choice for building native
+addons. However, Google has stopped development on GYP and some developers
+are concerned about the dependency of some GYP versions on Python 2.7 which
+has a sunset date of 2020.
```

Comment:
I don’t think the latter issue is a concern anymore if you’re using the latest versions of npm/node-gyp, so I’d omit both that and the statement that Google has stopped development – it’s not like we’re in a position to actually abandon support for node-gyp.

What you *could* mention is that developers have found node-gyp to have limitations. It’s somewhat opinionated, but I am sure many of us would agree with such a statement.

Reply:
Changed.